### PR TITLE
fix: remove duplicate language pill filtering logic

### LINKED ISSUE #1326: Language pills (multi-select) checked twice with conflicting logic branches
**Current Behavior**:
In `src/js/app.js` lines 542-543, language pill filtering is evaluated twice:
```javascript

if(pills.size>0){let m=false;pills.forEach(p=>{if(txt.includes(p))m=true;});if(!m)return false;}


if(pills.size>0&&!orgMatchesLanguages(o,pills))return false;
```

This creates two sequential filter conditions that operate on the same `pills` set:
1. **First condition** (line 542): Naive text matching — checks if org description includes raw pill label
2. **Second condition** (line 543): Semantic matching — uses `LANGUAGE_MAP` for aliases

**Scenario causing failure**:
- User selects "C/C++" pill
- Organization has `tags: ["c++", "python"]`
- First condition: Checks if txt includes raw "C/C++" → fails (org text has "c++" not "C/C++")
- Second condition: Correctly matches via `LANGUAGE_MAP["C/C++"] → ["c", "c++"]` → should pass
- **Result**: Org filtered out by first condition before second condition can run

**Expected Behavior**:
Language pill filtering should use consistent, semantic matching that respects the `LANGUAGE_MAP` and supports both AND/OR logic modes without duplicate evaluation.

**Technical Context**:
- **Affected files**: [src/js/app.js](src/js/app.js#L500-L550)
- **Root cause**: Refactoring artifact — likely when language alias support was added
- **Impact scope**: Affects all users with language filters; compound with search filters
- **Fix approach**: Remove the naive text-matching check (line 542) and rely solely on `orgMatchesLanguages()` which already handles OR/AND toggle logic correctly

---

### PR DESCRIPTION
**Pull request by PrincePundir123**
program: gssoc

## 📝 Description
This PR fixes a critical bug in language pill filtering where organizations with language tag aliases (e.g., "c++" for "C/C++") were being incorrectly filtered out. The issue was caused by duplicate, conflicting filter conditions that evaluated the same language filter twice with different matching strategies.

### Problem Description
In `src/js/app.js`, language pill filtering was evaluated twice with conflicting logic:
1. **Naive text matching**: Checked if the organization description includes the raw pill label (e.g. looks for literal "C/C++").
2. **Semantic matching**: Uses `LANGUAGE_MAP` for language aliases (e.g., resolves "C/C++" → `["c", "c++"]`).

Because the naive text matching ran first, organizations with tags like `["c++", "python"]` failed the literal "C/C++" check and were incorrectly filtered out before the semantic check could ever run.

### Fix Approach
Removed the naive text-matching check and now rely solely on `orgMatchesLanguages()`, which already:
- Properly resolves language aliases via `LANGUAGE_MAP`
- Supports both AND/OR logic modes via the `matchAllLanguages` toggle
- Correctly normalizes tags for comparison
- Eliminates unnecessary performance overhead from redundant loops

## 🔗 Related Issue
Closes #1326

## 🔄 Type of Change
- [x] 🐛 Bug fix
- [ ] ✨ New feature
- [ ] 🔍 SEO improvement
- [ ] 🎨 Style / UI improvement
- [ ] ♿ Accessibility improvement
- [ ] 📝 Documentation
- [ ] ⚙️ CI / configuration
- [ ] 🧹 Refactor / cleanup

## 🧪 How to Test
**Test Case 1: C/C++ Language Filter**
- Select the "C/C++" pill.
- Verify organizations with the "c++" tag appear (e.g., AFLplusplus, ArduPilot, Blender).
- Verify organizations without any C/C++ variant are hidden.

**Test Case 2: Multiple Language Aliases**
- JavaScript pill shows orgs with "js" tag.
- TypeScript pill shows orgs with "ts" tag.
- Go pill shows orgs with "golang" tag.
- ML/AI pill shows orgs with "machine learning", "ml", "ai" tags.

**Test Case 3: AND/OR Logic**
- Select multiple language pills (e.g., "Python" + "JavaScript").
- In OR mode: Shows orgs with EITHER language.
- Toggle to AND mode: Shows only orgs with BOTH languages.

**Test Case 4: Combined Filters & Edge Cases**
- Apply language filter + search filter / category filter simultaneously.
- Verify no unexpected filtering conflicts.
- Select all language pills (shows relevant orgs) and deselect all pills (shows all orgs).

## ✅ Checklist
- [x] I am contributing as an GSSoC contributor
- [x] My code follows the project's existing style
- [x] I have tested my changes in a browser
- [x] I have linked the related issue above
- [x] My PR title follows Conventional Commits format (e.g. `fix: remove duplicate language pill filtering logic`)
- [x] I have not introduced any new dependencies or build tools
- [x] Commits include DCO sign-off (`Signed-off-by:`)

### CODE REVIEW

**Review by Anushreebasics — DISMISSED** [review]
Thanks — the language-filter cleanup is good. A couple of fixes needed before approve: (1) escape org descriptions in the grid to prevent XSS (see [app.js:850]), (2) either sync [globalThis.matchAllLanguages] when the toggle changes or remove the global export ([src/js/app.js#L259], [src/js/app.js#L1364]). Also please add DCO sign-offs for the listed commits and address the auto-tools' suspicious-change notes (or explain them). After those, I'll re-review and approve.

**Comment by PrincePundir123** [comment]
@Anushreebasics Thanks for the feedback! I've fixed the XSS escape, synced the global toggle variable, signed the commit, and cleaned up the messy git history. Could you please take another look?

**Review by Anushreebasics — APPROVED** [review]
LGTM!

**Review by KumarNirupam1 — APPROVED** [review]
the language filter cleanup looks good now and the alias matching issue is properly fixed.

xss escape and toggle sync issues also seem resolved now.

LGTM

**Comment by Anushreebasics** [comment]
@PrincePundir123 pls resolve conflicts

File: src/js/app.js [hunk 1]
```diff
@@ -1363,6 +1364,7 @@ renderSelectedLanguages();
 // Initialize match mode toggle listener
 document.getElementById('matchAllLanguagesToggle')?.addEventListener('change', (e) => {
   matchAllLanguages = e.target.checked;
+  globalThis.matchAllLanguages = matchAllLanguages;
```
Comment (S3DFX-CYBER) [inline src/js/app.js:1367]: This `globalThis.matchAllLanguages` assignment is added here in app.js but is NOT mirrored in index.html's inline script. The test mirror and production runtime now diverge for filter logic. Both files must be updated together for any change to filter/sort/search behavior.

File: src/js/app.js [hunk 2]
```diff
@@ -847,7 +848,7 @@ function renderGrid(orgs){
           </a>`:''}
         </div>
       </div>
-      <div class="org-desc">${o.desc}</div>
+      <div class="org-desc">${escapeHtml(o.desc)}</div>
```
Comment (S3DFX-CYBER) [inline src/js/app.js:851]: Good XSS hardening — `escapeHtml(o.desc)`. But this same change must also be applied to the `renderGrid` function in index.html's inline script, where the production runtime lives. Currently only the test mirror is fixed.

**Review by S3DFX-CYBER — CHANGES_REQUESTED** [review]
This PR aims to fix a critical bug in language pill filtering by removing duplicate, conflicting filter conditions in `src/js/app.js`. The intent is solid — relying solely on `orgMatchesLanguages()` with `LANGUAGE_MAP` aliases is the right approach. However, there are a couple of blocking issues before this can merge.

The most important problem is the **frontend sync inconsistency**. The production runtime lives in the inline script of `index.html`, and `src/js/app.js` is its test mirror — both must stay in sync for any logic change. This PR adds `globalThis.matchAllLanguages = matchAllLanguages;` to the toggle listener in `app.js` (line ~1367) but does **not** make the corresponding change in `index.html`. The inline script's `matchAllLanguagesToggle` handler is left without this global assignment, meaning the production runtime and the test mirror now diverge. Any logic change to filter/sort/search behavior must be applied to both files.

Second, the PR body describes removing "the naive text-matching check" that looked for the literal pill label in the org description, but the diff only shows the comment update and the `orgMatchesLanguages()` call — there's no removal of the old naive matching block visible. If the duplicate condition was already cleaned up in a prior commit, the PR description should clarify that; otherwise the removal the PR claims to make isn't actually in the diff.

The `escapeHtml(o.desc)` change on line 851 is a welcome XSS hardening improvement, but again it's only applied in `app.js` and not mirrored in `index.html`'s inline `renderGrid`. If that function exists in the inline script with a raw `${o.desc}`, it needs the same escape treatment there.

Please sync all three changes (the global assignment, the filter logic, and the escapeHtml call) to `index.html`'s inline script so the production runtime matches the test mirror. Once that's done, this is good to go.
